fdbserver/Coordination.actor.cpp

```diff
@@ -776,3 +776,78 @@ ACTOR Future<Void> coordinationServer(std::string dataFolder,
 		throw;
 	}
 }
+
+ACTOR Future<Void> changeClusterDescription(std::string datafolder, KeyRef newClusterKey, KeyRef oldClusterKey) {
+	state UID myID = deterministicRandom()->randomUniqueID();
+	state OnDemandStore store(datafolder, myID, "coordination-");
+	RangeResult res = wait(store->readRange(allKeys));
+	// Context, in coordinators' kv-store
+	// cluster description and the random id are always appear together as the clusterKey
+	// The old cluster key, (call it oldCKey) below can appear in the following scenarios:
+	// 1. oldCKey is a key in the store: the value is a binary format of _GenerationRegVal_ which contains a different
+	// clusterKey(either movedFrom or moveTo)
+	// 2. oldCKey appears in a key for forwarding message:
+	// 		2.1: the prefix is _fwdKeys.begin_: the value is the new connection string
+	//		2.2: the prefix is _fwdTimeKeys.begin_: the value is the time
+	// 3. oldCKey does not appear in any keys but in a value:
+	// 		3.1: it's in the value of a forwarding message(see 2.1)
+	//		3.2: it's inside the value of _GenerationRegVal_ (see 1), which is a cluster connection string.
+	//			in particular, even we do not change it the cluster should still be good, but just to be safe here.
```

Comment:
> in particular, even we do not change it the cluster should still be good

Can you explain this a little more? It looks to me like the cluster key gets written at https://github.com/apple/foundationdb/blob/dc9599f2e907c2fb9d3c714684ed1ba8f92b251e/fdbserver/CoordinatedState.actor.cpp#L138, and then written into the `GenerationRegVal` at https://github.com/apple/foundationdb/blob/dc9599f2e907c2fb9d3c714684ed1ba8f92b251e/fdbserver/Coordination.actor.cpp#L145. Is the value not important for later reads?

Reply:
I am not 100% sure, but from my reading, the value here, a _MovableValue_, contains the previous change info.
In particular, it will contain the connection string it changed from or changed to. But as the change has happened, the new cluster key will appear in the kv store, which we already changed and is enough for the coordinators?
It will be more clear once I finish the test to run this automatically in restarting tests.
For now, I just change any old cluster key to the new cluster key.